Log In + Sign Up

brigittewylee opened the pull request:
adds the Login Page:

<img width="1440" alt="image" src="https://github.com/Your-Missing-Semester/WayPoints/assets/150951722/af8628ac-39d7-446e-af12-b9d2ba576a7e">

chinkevin25 commented on frontend/src/components/LogIn/login.css:
```diff
@@ -118,3 +128,34 @@ img {
+.reset {
+    text-decoration: underline;
+}
+.reset:hover{
```
Same kind of comments as in your Other PR. 

Lets look into modules as well as making sure we format our code consistently.

chinkevin25 commented on frontend/src/components/LogIn/login_error.jsx:
Same comments as your signup error page!

chinkevin25 commented:
Oops forgot to comment a few things.

I noticed that the diff between this PR and the other PR (signup) were very very similar so I changed your base branch from main to `brigitte/SignUp`. This just lets me see the differences between the two branches. A lot of the comments from the SignUp branch applies here too.

Your PR description only shows a picture of the SignUp page, but technically, this PR is for the Login Page.

brigittewylee commented:
**Fixed:**
- Formatting (i.e. whitespace, snake case in css, deleted unused code)
- Changed to css modules
- Removed error page and added conditional rendering components
- Added react router between pages (updates to app.js in sign up pr)
- Separated oauth logos to own folder

**Updates:**
- Regex for email validity
- Changed from sign in to login for clarity
- Added error message for password validity for later use (comparison to user set passwords)
- Changed sign up button to link

<img width="1421" alt="image" src="https://github.com/Your-Missing-Semester/WayPoints/assets/150951722/f117b2a1-261e-4627-8c03-bfe5f9a8b603">

chinkevin25 commented on package.json:
can we delete this file I think we npm init on accident at the root

chinkevin25 commented on package-lock.json:
same here, i don't think we need a root level package json yet

chinkevin25 commented on frontend/src/components/SignUp/signup.module.css:
```diff
@@ -0,0 +1,131 @@
+    margin-bottom: 3%;
+}
+
+.signup_button{
```
same comment here where we have inconsistent css styling here. There's no space after the css selector, but there are in other selectors

chinkevin25 commented on frontend/src/components/SignUp/signup.module.css:
```diff
@@ -0,0 +1,131 @@
+    margin-bottom: 0px;
+}
+
+.create_account_subtext{
```
I must have forgotten to mention this but generally css classes are kabob cased `create-account-subtext`. It does make accessing the CSS module style a little clunkier (e.g. `styles['create-account-subtext']`) but there are ways to fix that as well

chinkevin25 commented on frontend/src/components/SignUp/signup.jsx:
```diff
@@ -0,0 +1,89 @@
+import twtx_icon from '../Logos/twtx_icon.webp';
+import apple_icon from '../Logos/apple_icon.png';
+
+function EmailErrorMessage({username, isValid}) {
+    return <p className={styles.error_message}>Please enter a valid email.</p>
+}
+
+function PasswordErrorMessage({password, isValid}) {
+    return <p className={styles.error_message}>Passwords must be at least 8 characters and 
+    include at least one lowercase, uppercase, number, and special symbol (~|!@#$%^&*).</p>
+}
+
+function PasswordConfirmation({password, isValid}) {
+    return <p className={styles.error_message}>Passwords do not match. </p>
+}
```
No change needed -- this code will probably change once we do proper server authentication but, 
A couple nits here:

1. in general we strive to do 1 component per file do these should be in its own file
2. We could create a standardize "error message" component if we wanted to. Right now these are all the same basically, it takes it a boolean, the other argument isn't even being used.

chinkevin25 reviewed (commented):
Just some superficial changes! We can merge this soon!!

fahadakhan2 commented on frontend/src/App.js:
```diff
@@ -1,23 +1,17 @@
+      <Router>
+          <Routes>
+              <Route exact path="/" element={<h1>Home Page</h1>} />
+              <Route exact path="/SignUp" element={<SignUp />} />
```
For paths we want to use lower kebab case. Please change the path to to "/sign-up".

fahadakhan2 commented on frontend/src/App.js:
```diff
@@ -1,23 +1,17 @@
+          <Routes>
+              <Route exact path="/" element={<h1>Home Page</h1>} />
+              <Route exact path="/SignUp" element={<SignUp />} />
+              <Route exact path="/LogIn" element={<LogIn />} />
```
Same here but since login is one word you can change the path to "/login"

fahadakhan2 commented on frontend/src/components/LogIn/login.jsx:
For components we want to name the file upper camel case.

Change to Login.jsx

fahadakhan2 commented on frontend/src/components/LogIn/login.module.css:
Same thing here since we are upper camel casing our component file names we want to uppercamel case the prefix of the file extension .module.css.

This file should be named Login.module.css

fahadakhan2 commented on frontend/src/components/LogIn/login.jsx:
```diff
@@ -0,0 +1,56 @@
+import twtx_icon from '../Logos/twtx_icon.webp';
+import apple_icon from '../Logos/apple_icon.png';
+
+function EmailErrorMessage({username, isValid}) {
```
For functions in javascript (not including naming the functional component) we use camel case.

Rename function to emailErrorMessage

chinkevin25 replied:
Technically this is a JSX React Component, so PascalCase is correct. But see my comment on this stuff, they should technically be in separate files, but i think we can hold off on that since this form will probably need to change when we do proper auth

fahadakhan2 commented on frontend/src/components/LogIn/login.jsx:
```diff
@@ -0,0 +1,56 @@
+//     return <p className={styles.error_message}>Incorrect password </p>
+// }
+
+export function LogIn() {
```
login is one word so no need to upper camel case here

Change LogIn to Login

fahadakhan2 commented on frontend/src/components/LogIn/login.jsx:
```diff
@@ -0,0 +1,56 @@
+            <h1>LOGIN</h1>
+            <form>
+                <label>Email</label>
+                    <input type="text" placeholder="Email"onChange={(event)=>{
```
Change the type attribute to be "email".

Modern browsers automatically perform basic email validation on inputs with type="email". This helps ensure users enter valid email addresses.

fahadakhan2 replied:
For the onChange function you should extract the logic into a seperate function outside of the jsx. Then call that function by just passing it the name and invoking it in the handler. Makes the jsx a lot more readable.

fahadakhan2 commented on frontend/src/components/LogIn/login.jsx:
```diff
@@ -0,0 +1,56 @@
+import twtx_icon from '../Logos/twtx_icon.webp';
+import apple_icon from '../Logos/apple_icon.png';
+
+function EmailErrorMessage({username, isValid}) {
```
If this is a component you should extract the logic into its own file under the components folder.

fahadakhan2 commented on frontend/src/components/LogIn/login.jsx:
```diff
@@ -0,0 +1,56 @@
+                    {!isEmailValid && <EmailErrorMessage></EmailErrorMessage>}
+
+                <label>Password</label>
+                    <input type="text" placeholder="Password"/>
```
You can change the attribute type to "password".  The characters typed into a password input are masked (typically with dots or asterisks) to prevent others from seeing the password.

fahadakhan2 commented on frontend/src/components/LogIn/login.jsx:
```diff
@@ -0,0 +1,56 @@
+                <label>Password</label>
+                    <input type="text" placeholder="Password"/>
+
+                    <p className={styles.pass_reset}>Forgot Password?</p>
```
Right now this is just text. But it would be better to make it a Link because when the user clicks on this it would take the user to a /forgot-password route. Even if we don't have the UI for that route yet it's still good to have it early on.

fahadakhan2 commented on frontend/src/components/LogIn/login.jsx:
```diff
@@ -0,0 +1,56 @@
+                <button className={styles.login_button}>LOGIN</button>
+            </form>
+            <p className={styles.signup_text}> Don't have an account?
+            <Link to="/SignUp" className={styles.signup_link}>Join us today!</Link></p>
```
will have to change to attribute to "/sign-up"

fahadakhan2 commented on frontend/src/components/LogIn/login.module.css:
```diff
@@ -0,0 +1,152 @@
+    padding-bottom: 2%;
+}
+
+.login_group {
```
For all css the selectors should be lower kebab cased.

For example this should be .login-group {...}

fahadakhan2 commented on frontend/src/components/Logos/apple_icon.png:
1. All these images that you are using should be put into a "assets" folder at the same level your components folder sits on in the file structure.
2. lower camel case your image files. For example this one would be appleIcon.png

fahadakhan2 commented on frontend/src/components/SignUp/signup.jsx:
```diff
@@ -0,0 +1,89 @@
+
+export function SignUp() {
+
+    // notes:
```
Since we are going to be merging this code into main we don't want a big chunk of comments in our codebase. I'd highly recommend just making a google doc to put down any notes you need for the future. But please remove these comments for the PR.

fahadakhan2 commented on frontend/src/components/SignUp/signup.jsx:
```diff
@@ -0,0 +1,89 @@
+
+    const [email, setEmail]=useState('')
+    var checkEmail= new RegExp("^.+@.+\\..+")
+    // checks email for @ and subsequent . (might have to update to make more accurate later)
```
remove comment

fahadakhan2 commented on frontend/src/components/SignUp/signup.jsx:
```diff
@@ -0,0 +1,89 @@
+
+    const [password, setPassword]=useState('')
+    var passRegex=new RegExp("^(?=.*[a-z])(?=.*[A-Z])(?=.*\\d)(?=.*[@$!%*?&])[A-Za-z\\d@$!%*?&]{8,}$")
+    // regex to meet min 8 characters consisting of at least one lowercase, uppercase, number, and special symbol
```
remove comment

chinkevin25 replied:
I would probably keep this comment at the very least b/c it's commenting what the regex is doing.

fahadakhan2 commented on frontend/src/components/SignUp/signup.jsx:
```diff
@@ -0,0 +1,89 @@
+    const [password, setPassword]=useState('')
+    var passRegex=new RegExp("^(?=.*[a-z])(?=.*[A-Z])(?=.*\\d)(?=.*[@$!%*?&])[A-Za-z\\d@$!%*?&]{8,}$")
+    // regex to meet min 8 characters consisting of at least one lowercase, uppercase, number, and special symbol
+    const isPasswordValid=password.length===0 || passRegex.test(password)
```
Might want extract this type of logic into a function.

fahadakhan2 commented on frontend/src/components/SignUp/signup.jsx:
```diff
@@ -0,0 +1,89 @@
+            <p className={styles.create_account_subtext}>create account</p>
+            <form>
+                <label>Email</label>
+                    <input type="text" placeholder="asdf@gmail.com" onChange={(event)=>{
```
Change attribute type to "email"

fahadakhan2 replied:
Change the placeholder to "Email"

fahadakhan2 replied:
Extract the onChange arrow function into its own function and pass in the name afterwards.

fahadakhan2 commented on frontend/src/components/SignUp/signup.jsx:
```diff
@@ -0,0 +1,89 @@
+                    {!isEmailValid && <EmailErrorMessage></EmailErrorMessage>}
+
+                <label>Password</label>
+                    <input type="text" placeholder="Password" onChange={(event)=>{
```
Extract the onChange arrow function into its own function and pass in the name afterwards.

fahadakhan2 commented on frontend/src/components/SignUp/signup.jsx:
```diff
@@ -0,0 +1,89 @@
+                    {!isPasswordValid && <PasswordErrorMessage></PasswordErrorMessage>}
+
+                <label>Confirm Password</label>
+                    <input type="text" placeholder="Password" onChange={(event)=>{
```
The placeholder here makes more sense to be Confirm Password

fahadakhan2 commented on frontend/src/components/SignUp/signup.jsx:
```diff
@@ -0,0 +1,89 @@
+                <button className={styles.signup_button}>SIGN UP</button>
+            </form>
+            <p className={styles.login_subtext}>Already have an account?  
+            <Link to="/LogIn" className={styles.login_button}>Log in here</Link></p>
```
to attribute should be changed to /login

fahadakhan2 reviewed (changes requested):
Left some more comments on top of Kevins.
